src/librustc_data_structures/obligation_forest/mod.rs

```diff
@@ -287,14 +402,30 @@ impl<O: ForestObligation> ObligationForest<O> {
     pub fn new() -> ObligationForest<O> {
         ObligationForest {
             nodes: vec![],
-            done_cache: Default::default(),
+            pending_nodes: vec![],
+            success_or_waiting_nodes: vec![],
+            error_or_done_nodes: RefCell::new(vec![]),
+            dead_nodes: vec![],
             active_cache: Default::default(),
-            node_rewrites: vec![],
             obligation_tree_id_generator: (0..).map(ObligationTreeId),
+            node_number: 0,
             error_cache: Default::default(),
+            stalled_on: Default::default(),
+            unblocked: Default::default(),
+            stalled_on_unknown: Default::default(),
+            temp_unblocked_nodes: Default::default(),
+            offset: None,
         }
     }
 
+    pub fn offset(&self) -> Option<&O::WatcherOffset> {
```

Comment:
I think it should have a doc comment, though it could "forward" to the offset field ---

Return the offset registered with notification table. See the field `ObligationForest::offset`.